docs/decisions/0005-event-schema-serialization-and-evolution.rst

```diff
@@ -0,0 +1,75 @@
+
+Open edX is currently experimenting with event bus technology (`OEP-52: Event Bus <https://github.com/openedx/open-edx-proposals/pull/233>`_). The goal is to iterate through different implementations of an event bus and slowly get to a solution that works for the larger Open edX developer community.
+
+Currently, the specification for an event are written as OpenEdxPublicSignal instances in signal.py modules in `openedx-events <https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/learning/signals.py>`_ repository.
+
+Over time, as needs change, applications will need the schema for an event to change.
+
+For internal events, there is some discussion of versioning events related to how they might change in https://github.com/eduNEXT/openedx-events/blob/main/docs/decisions/0002-events-naming-and-versioning.rst. Schema for event bus messages is defined in openedx-events as signal classes.
+
+However, for the event bus, we will be introducing more rigorous event evolution rules using an explicit schema and schema registry. We have decided to use Avro elsewhere (`OEP-52/decisions/001-schema-representation.rst <https://github.com/openedx/open-edx-proposals/pull/233/files#diff-70c71499189a23f546da507c5bfdc0fea674f4cbbc5c039298d8390d6930a5ca>`_).
+
+Event schemas can evolve in various ways and each implementation of an event bus should choose a schema evolution configuration. The choices include how data can change, as well as the order in which producer and consumer needs to be deployed with new changes.
+
+Event bus technologies support various schema evolutions::
+
+- Backward / Backward Transitive: Allows you to delete fields and add optional fields. The consumer of the messages must upgrade to handle new schema before producer.
+
+- Forward / Forward Transitive: Allows you to add fields and delete optional fields. The producer of the messages must upgrade to handle new schema before consumer of  messages.
+
+- Full / Full Transitive: Allows you to add or delete optional fields. Either producer or consumer can change first.
+
+
+Currently, the schema for event bus messages in defined through two classes: OpenEdxPublicSignal and attrs decorated classes in data.py modules (`example <https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/learning/data.py>`_).
+
+OpenEdxPublicSignal defines all the information necessary to create, serialize, and send an event. This includes the data to be sent and the metadata. The data to be sent is defined as the 'init_data' attribute. 'init_data' is a dict of key/value pairs. The keys are strings and the values can be of various types (the exact supported types has not be specified yet). These keys correspond to keyword arguments sent via the signal.
+
+The attrs decorated classes are used to structure the data sent. Hopefully, most of the values in 'init_data' attribute in OpenEdxPublicSignal are of these attrs decorated classes.
```

Comment:
```suggestion
At this time, most of the values in `init_data` are special attr decorated classes used to provide additional structure and reuse of the data.
```
Also, add a link to the data modules from above if you wish.

Reply:
I would like to keep the language to suggest that most of the info in signals should be in form of attrs decorated classes. I'd like steer people towards attrs classes as much as possible.